### PR TITLE
#411@minor: Adds support for document.getElementsByName

### PR DESCRIPTION
**Pull request by Mas0nShi**
solve #411

### CODE REVIEW

**Comment by Mas0nShi** [comment]
need more test units …

File: packages/happy-dom/src/nodes/element/Element.ts [hunk 1]
```diff
@@ -796,6 +796,15 @@ export default class Element extends Node implements IElement {
...
 
+	/**
+	 * Returns an elements by name.
+	 *
+	 * @returns Matching element.
+	 * @param name
+	 */
+	public getElementsByName(name: string): IHTMLCollection<IElement> {
+		return ParentNodeUtility.getElementsByName(this, name);
+	}
```
Comment (capricorn86) [inline packages/happy-dom/src/nodes/element/Element.ts:807]:
```suggestion
```

Element.getElementsByName() is not part of the HTML specification.

File: packages/happy-dom/src/nodes/parent-node/ParentNodeUtility.ts [hunk 2]
```diff
@@ -190,4 +190,29 @@ export default class ParentNodeUtility {
...
 	}
+
+	/**
+	 * Returns an element by Name.
+	 *
+	 * @param parentNode
+	 * @param name Name.
+	 * @returns Matching element.
+	 */
+	public static getElementsByName(
```
Comment (capricorn86) [inline packages/happy-dom/src/nodes/parent-node/ParentNodeUtility.ts:201]: This function can be moved to Document.getElementsByName().

File: packages/happy-dom/test/nodes/parent-node/ParentNodeUtility.test.ts [hunk 3]
```diff
@@ -233,4 +233,13 @@ describe('ParentNodeUtility', () => {
 			expect(ParentNodeUtility.getElementById(parent, 'div3')).toEqual(div3);
 		});
 	});
+
+	describe('getElementsByName()', () => {
```
Comment (capricorn86) [inline packages/happy-dom/test/nodes/parent-node/ParentNodeUtility.test.ts:237]: This test can be moved to Document.test.ts

File: packages/happy-dom/src/nodes/parent-node/ParentNodeUtility.ts [hunk 4]
```diff
@@ -190,4 +190,29 @@ export default class ParentNodeUtility {
...
+		parentNode: IElement | IDocumentFragment | IDocument,
+		name: string
+	): IHTMLCollection<IElement> {
```
Comment (capricorn86) [inline packages/happy-dom/src/nodes/parent-node/ParentNodeUtility.ts:204]:
```suggestion
	): INodeList<IElement> {
```

File: packages/happy-dom/src/nodes/document/Document.ts [hunk 5]
```diff
@@ -338,6 +338,16 @@ export default class Document extends Node implements IDocument {
 		return <Element>ParentNodeUtility.getElementById(this, id);
 	}
 
+	/**
+	 * Returns an element by Name.
+	 *
+	 * @returns Matching element.
+	 * @param name
+	 */
+	public getElementsByName(name: string): IHTMLCollection<IElement> {
```
Comment (capricorn86) [inline packages/happy-dom/src/nodes/document/Document.ts:347]:
```suggestion
	public getElementsByName(name: string): INodeList<IElement> {
```

**Review by capricorn86 — CHANGES_REQUESTED** [review]

File: packages/happy-dom/src/nodes/parent-node/ParentNodeUtility.ts (hunk 2 above)
Comment (Mas0nShi) [reply packages/happy-dom/src/nodes/parent-node/ParentNodeUtility.ts:202]: recursion function, i will try to move it in my power.

**Review by Mas0nShi — COMMENTED** [review]
ok, plz review again...

**Review by capricorn86 — APPROVED** [review]
Nice work! :star2:

**Pull request merged**